state-chain/custom-rpc/src/lib.rs

```diff
@@ -1744,6 +1851,212 @@ where
+	#[method(name = "authorities")]
+	fn cf_authorities(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<AuthoritiesInfo>;
+	#[method(name = "external_chains_block_height")]
+	fn cf_external_chains_block_height(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<ExternalChainsBlockHeight>;
+	#[method(name = "btc_utxos")]
+	fn cf_btc_utxos(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<BtcUtxos>;
+	#[method(name = "dot_aggkey")]
+	fn cf_dot_aggkey(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<PolkadotAccountId>;
+	#[method(name = "suspended_validators")]
+	fn cf_suspended_validators(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<Vec<(Offence, u32)>>;
+	#[method(name = "epoch_state")]
+	fn cf_epoch_state(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<EpochState>;
+	#[method(name = "redemptions")]
+	fn cf_redemptions(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<RedemptionsInfo>;
+	#[method(name = "pending_broadcasts")]
+	fn cf_pending_broadcasts(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<PendingBroadcasts>;
+	#[method(name = "pending_tss_ceremonies")]
+	fn cf_pending_tss_ceremonies(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<PendingTssCeremonies>;
+	#[method(name = "pending_swaps")]
+	fn cf_pending_swaps(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<u32>;
+	#[method(name = "open_deposit_channels")]
+	fn cf_open_deposit_channels(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<OpenDepositChannels>;
+	#[method(name = "fee_imbalance")]
+	fn cf_fee_imbalance(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<FeeImbalance>;
+	#[method(name = "build_version")]
+	fn cf_build_version(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<LastRuntimeUpgradeInfo>;
+	#[method(name = "data")]
+	fn cf_monitoring_data(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<RpcMonitoringData>;
+	#[method(name = "accounts_info")]
+	fn cf_accounts_info(
+		&self,
+		accounts: BoundedVec<state_chain_runtime::AccountId, ConstU32<10>>,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<Vec<RpcAccountInfoV2>>;
+}
+
+impl<C, B> MonitoringApiServer for CustomRpc<C, B>
+where
+	B: BlockT<Hash = state_chain_runtime::Hash, Header = state_chain_runtime::Header>,
+	C: sp_api::ProvideRuntimeApi<B>
+		+ Send
+		+ Sync
+		+ 'static
+		+ HeaderBackend<B>
+		+ BlockchainEvents<B>,
+	C::Api: MonitoringRuntimeApi<B>,
+{
+	fn cf_authorities(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<AuthoritiesInfo> {
+		self.client
+			.runtime_api()
+			.cf_authorities(self.unwrap_or_best(at))
+			.map_err(to_rpc_error)
+	}
+	fn cf_external_chains_block_height(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<ExternalChainsBlockHeight> {
+		self.client
+			.runtime_api()
+			.cf_external_chains_block_height(self.unwrap_or_best(at))
```

Comment:
Most of these methods are just passed through to the runtime api.

We could write a simple macro that reduces the repetition here. 

Not sure about the exact syntax but something like this:

```rust
macro_rules! pass_through {
	$( $name:ident -> $result_type:ty ),+
} {
	$(
		fn $name(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<$result_type> {
			self.client
				.runtime_api()
				.$name(self.unwrap_or_best(at))
				.map_err(to_rpc_error)
		}
	)+
}
```

Would let us write this for all of the simple cases: 

```
pass_through! {
	cf_authorities -> AuthoritiesInfo,
	cf_external_chains_block_height -> ExternalChainsBlockHeight,
	// ...
}
```